src/components/header/Backtrace.js

```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import React, { PureComponent } from 'react';
+import { filterCallNodePathByImplementation } from '../../profile-logic/transforms';
+import { getOriginAnnotationForFunc } from '../../profile-logic/profile-data';
+
+import type { Thread, IndexIntoStackTable } from '../../types/profile';
+import type { CallNodePath } from '../../types/profile-derived';
+import type { CauseBacktrace } from '../../types/markers';
+import type { ImplementationFilter } from '../../types/actions';
+
+require('./Backtrace.css');
+
+type Props = {
+  thread: Thread,
+  cause: CauseBacktrace,
+  implementationFilter: ImplementationFilter,
+};
+
+type State = {};
+
+function callNodePathForStack(
+  thread: Thread,
+  stack: IndexIntoStackTable
+): CallNodePath {
+  const { stackTable, frameTable } = thread;
+  const path = [];
+  for (
+    let stackIndex = stack;
+    stackIndex !== null;
+    stackIndex = stackTable.prefix[stackIndex]
+  ) {
+    path.push(frameTable.func[stackTable.frame[stackIndex]]);
+  }
+  return path;
+}
+
+class Backtrace extends PureComponent<Props, State> {
+  render() {
+    const { cause, thread, implementationFilter } = this.props;
+    const { funcTable, stringTable } = thread;
+    const callNodePath = filterCallNodePathByImplementation(
+      thread,
+      implementationFilter,
+      callNodePathForStack(thread, cause.stack)
+    );
```

Comment:
sometimes it's empty here when we use the JS-only implementation filter; should we display something in that case? Maybe we shouldn't display the `at` string in that case too, so maybe move it here ? Or move this call to the parent component ? I don't know :)
This would get better once we have pseudoframes.

Also are we getting the proper stack for everything done by JS? I'm thinking for example of when we switch a class, then a restyle is performed. In that case, I we'd see a JS stack for the 'styles' marker, and another one for the 'reflow' marker?